SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+LSST nomenclature is to use \emph{PARENT} to indicate the abstract pixel coordinate and \emph{LOCAL} to indicate that the coordinates are relative to the 0-based origin of the data array itself\footnote{The Starlink nomenclature\cite{2015A&C....12..146J} is \emph{PIXEL} and \emph{GRID} where \emph{GRID} follows the FITS convention of starting at 1.}.
+This attachment of a data array to a parent pixel coordinate system is a critical item and we plan to work with Astropy on making this functionality available to \texttt{NDData} users.
+
+\begin{figure} [t]
+\begin{center}
+\includegraphics[width=\textwidth]{exposure-dependencies}
+\end{center}
+\caption[layers]
+%>>>> use \label inside caption to get Fig. number with \ref{}
+{\label{fig:exposure}
+The LSST classes that are used to define a \texttt{lsst.afw.image.Exposure}.}
+\end{figure}
+
+\section{Suggested modifications to LSST}
+
+In addition to working with the Astropy community, there is also the question of how integrated the LSST software system should be with Astropy.
+This covers the adoption of distribution standards and use of Astropy classes in our codebase.
+These changes will result in a more supportable and sustainable codebase allowing LSST to focus on core deliverables without having to support general infrastructure.
+
+\subsection{Packaging}
+
+The LSST data management software uses the EUPS tool\cite{EUPS} to manage versions and dependencies.
+It is used to ensure that the versions of all packages used for a particular processing run can be assigned a single number and also to track the packages that were part of a single build.
+Development versions can easily be included along with versions from builds done in the past.
+This can significantly simplify debugging cycles and aid with provenance tracking but it is a complex environment with a steep learning curve and most users of LSST software outside of the data center and LSST developer community do not need this power.
+
+LSST packages are built using the SCons tool\footnote{\url{http://www.scons.org}}\cite{2005Scons1377085}.
+SCons is an alternative tool to Make and CMake that is written entirely in Python.
+This allows developers to use Python for their library code and build system whilst making use of the powerful source code dependency tracking provided by SCons to prevent wasteful recompilation of files that have not been affected by local edits.
+
+Unfortunately for SCons the majority of the Python community have adopted a simpler build system based around the convention of a \texttt{setup.py} file that can be executed to do the build.
+This file uses \texttt{distuils} or \texttt{setuptools} to determine what should be built and can handle most cases where external C/C++ code should be compiled into the Python package.
```

Comment:
In theory I imagine it could start a scons subprocess. `setup.py` can do whatever you want so long as it know how to handle "build", "test" and "install" targets (although there are quite a lot of verbs that people expect to work with `setup.py` such as "bdist upload"). At least now it is possible to `pip install scons` (a fairly new development) so it could add a `scons` dependency (and `eups`) using setuptools.. It seems a bit weird doing that though.